### ISSUE TITLE
Set FET tallies in OpenMC based on incoming info from MOOSE

### ISSUE THREAD

**Issue by aprilnovak** [aprilnovak/OkapiMCS#14]
It would be very convenient if we were able to set up the FETs within OpenMC directly based on information coming from MOOSE. We can use functions in `openmc/src/tallies/tally_header.F90` such as `openmc_extend_tallies` and `openmc_tally_set_filters` to set up tallies in the global `tallies` array for cells and expansion types specified in MOOSE. 

We can keep the older method of independently specifying the FETs in MOOSE and OpenMC, but because this can be cumbersome if you want to change the orders, this newer approach would be more elegant.

**Comment by paulromano** [comment]
Agreed -- this is the approach we will be taking for our internal driver. I'm planning on spending some time this week to get a pull request together for the FE filter branch, so I'll also pay attention to any specific capabilities that would be needed in the C API for FE filters.
